### DISCUSSION TITLE
OpenROAD build with recent versions of FMT and SPDLOG

### DISCUSSION THREAD

**Discussion by sebinho** [The-OpenROAD-Project/OpenROAD#5101] [General]
I am not sure where to put this, but I had lots of issues lately building OpenROAD on ArchLinux and I thought this might help other people.
The issues I had seem to be related to more recent versions of spdlog and fmt packages.
Dowgrading fmt to 8.1.1-2 and spdlog to 1.13.0 solves the issues (latest versions are 10.2.0-1 and 1.14.1-1).

The issues were as follows during the build (one example, others are similar):

```
/usr/include/fmt/core.h:1600:63: error: ‘fmt::v10::detail::type_is_unformattable_for<const dpl::TypedCoordinate<dpl::GridYType>, char> _’ has incomplete type
 1600 |     type_is_unformattable_for<T, typename Context::char_type> _;
      |                                                               ^
/usr/include/fmt/core.h:1604:7: error: static assertion failed: Cannot format an argument. To make type T formattable provide a formatter<T> specialization: https://fmt.dev/latest/api.html#udt
 1604 |       formattable,
      |       ^~~~~~~~~~~
/usr/include/fmt/core.h:1604:7: note: ‘formattable’ evaluates to false
```

The better way for me to solve my issue (instead of downgrading packages) was to build OpenROAD with the following flag during `cmake`: `-DSPDLOG_FMT_EXTERNAL=OFF`

**Comment by rovinski** [comment]
You might want to take a look at the dependency installer because this issue is known. It may not have support for Arch but you can look at comments for other distros for notes like this.

**Comment by maliberty** [comment]
Please see my comment about FMT_DEPRECATED_OSTREAM  in https://github.com/The-OpenROAD-Project/OpenROAD/pull/2696
